dev/tools/pre-commit

```diff
@@ -4,6 +4,24 @@
 
 set -e
 
+CODE=0
+for f in $(git diff --cached --name-only)
```

Comment:
This isn't robust wrt filenames with spaces/newlines/glob characters...

Instead we should use something like
```sh
# xargs exit status is 123 if any file failed the test
git diff --cached --name-only -z | xargs -0 dev/tools/fix-eof-newline.sh || CODE=1
```

This means `fix-eof-newline.sh` should handle multiple files (or use
-L 1 in the xargs) and exit 1 if there are errors. It also needs to
inline `should-check-whitespace.sh`.

With c671d86 it would just mean adding an option `--fix` to
`check-eof-newline.sh`, but that's still in the future.

Reply:
Thanks, fixed